src/System.Private.CoreLib/src/System/HighPerformanceCounter.cs

```diff
@@ -0,0 +1,42 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Runtime.CompilerServices;
+
+namespace System
+{
+    internal static class HighPerformanceCounter
+    {
+        [MethodImpl(MethodImplOptions.InternalCall)]
+        internal static extern ulong QueryPerformanceFrequency();
```

Comment:
It would be nice to wait for @AaronRobinsonMSFT #26458 to go through and the implement it using that.

Reply:
It looks like #26458 is about PInvoking to external libraries. I need to call corlib since it shims these methods for multiple platforms (Linux, OSX).

I am wondering if #26458 will help this case.

Reply:
It would use corefx shims, like CoreRT and Stopwatch does today.

Reply:
Coreclr needs these APIs and will have to shim for any new platform added. Why not just use that?

Reply:
The code written against CoreFX shims is shareable between CoreCLR, Mono and Corert. We are trying to maximize amount of such shared code,

Reply:
Also  #26458 seems to be inserting `GCPoll` in every call. Is that a good idea for a high performance counter?

Reply:
Yes, I think we need to start working towards more predictable GC suspension schemes to support GC SLAs. Many FCalls are missing the GC poll today - it is a bug.

Reply:
Just to be clear - I am not arguing against #26458 in general. I like it and would use it in a many other places. 
But for high performance counter having as little overhead as possible is important. FCall is the best option that we have for low overhead. I think it is more justified here than in many other places where FCall is used.

Reply:
FCalls without a GC poll are also a reliable way to stall the GC. It is not hard to come up with FCall-heavy microbenchmarks that have GC suspension times 0.5 seconds and more.

The two instructions one pays for the GC poll is absolutely worth paying to mitigate this problem. I am expect that we may enable the GC poll across the board for all FCalls once we start being serious about GC SLAs.

Note you have to pay the cost of the API wrapper with FCall. #26458 lets you call the OS API directly (on Windows at least), and so it is going to faster than what you got here, even with GC poll.